### DISCUSSION TITLE
WETH from Solana to ETH

### DISCUSSION THREAD

**Discussion by Bitclapp** [wormhole-foundation/wormhole#1013] [Problems / bugs] [answered]
Hi all,
I used the website https://v1.wormholebridge.com/ to transfer 0.5 WETH (wormhole) from Solana to the ETH main net
The transaction was running for the longest time and somehow it took my funds but never gave me an option to redeem on the ETH side.
Can you advise on the best way to address this?
My transaction is: https://solscan.io/account/9zyPU1mjgzaVyQsYwKJJ7AhVz5bgx5uc1NPABvAcUXsT

Any help is greatly appreciated.

Thanks

**Comment by evan-gray** [comment]
There is a third-party recovery tool that you could try (at your own risk, no warranties, etc) - https://www.interstellar.tk/
Please note that the v1 will be decommissioned on March 31st. 
https://twitter.com/wormholecrypto/status/1498699067374940162

**Reply by Bitclapp** [reply]
Thanks.  I tried it already and got some errors that can be seen in this issue I raised : https://github.com/solana-labs/oyster/issues/530 
Not sure what the next step would be... anything else I can try?

**Reply by Bitclapp** [reply]
@evan-gray Is there anything else I can do to recoup my funds... I'm really not in a position to be losing this money.  I made a terrible noob mistake by using v1  which I got from Google... but I'm really hoping there's something I can do to remedy the situation.  Any additional help is appreciated!  Thanks.

**Reply by evan-gray** [reply]
In that screenshot it said transaction failed but didn’t capture the reason. Any chance you could share the full error? I will also ask one of the V1 experts to take a look.

**Reply by Bitclapp** [reply]
Hi @evan-gray, sure I'm attaching 2 screenshots covering the entire error message.  I just tried it again now.  Thanks for following up.  

![Error_Pt1](https://user-images.githubusercontent.com/101778179/158912998-1a6815dd-4200-40bc-b83d-e2880ec06a3d.jpg)
![Error_Pt2](https://user-images.githubusercontent.com/101778179/158913004-2b6fa3e2-9b00-4793-b156-499e0c224363.jpg)

**Reply by SEJeff** [reply]
Any chance you can copy and paste that content into this issue? It makes it hard to troubleshoot if we have to type it all in manually.

**Reply by Bitclapp** [reply]
Sure, will do now... thanks!

**Reply by Bitclapp** [reply]
3/17/2022, 7:42:54 PM - Analyzing transaction "KTdkVHDwTCcDk6ptqEGDfQ8sQhY9HnjqSSuhopkz2BgBrdAnfJjhf6CeGiHFDVzAypckG7sMMrhonDtdWM8DbPv"
3/17/2022, 7:42:55 PM - Fetch transaction info successfully
3/17/2022, 7:42:55 PM - Transfer Amount: 0.50484747
3/17/2022, 7:42:55 PM - Target Network: Ethereum
3/17/2022, 7:42:55 PM - Target Address: 0X80253F5106F929D08AB0F69679284AE8A0CCA51C
3/17/2022, 7:42:55 PM - Extract VAA: 00000000000100736f6c616e615f73797374656d5f70726f6772616d0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
3/17/2022, 7:42:55 PM - Creating TX to mint asset
3/17/2022, 7:43:09 PM - Waiting TX to be mint
3/17/2022, 7:43:58 PM - Error: transaction failed (transactionHash="0x9ce3bc6f9a3cff3754254cba691d14728f16bd753d46566dbb96711e5c560f8e", transaction={"hash":"0x9ce3bc6f9a3cff3754254cba691d14728f16bd753d46566dbb96711e5c560f8e","type":2,"accessList":[],"blockHash":null,"blockNumber":null,"transactionIndex":null,"confirmations":0,"from":"0x80253f5106f929d08ab0f69679284ae8A0cCA51c","gasPrice":{"type":"BigNumber","hex":"0x0cced9fc80"},"maxPriorityFeePerGas":{"type":"BigNumber","hex":"0x5bca8900"},"maxFeePerGas":{"type":"BigNumber","hex":"0x0cced9fc80"},"gasLimit":{"type":"BigNumber","hex":"0x03d090"},"to":"0xf92cD566Ea4864356C5491c177A430C222d7e678","value":{"type":"BigNumber","hex":"0x00"},"nonce":2,"data":"0x3bc0aee6000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000003ed00000000000100736f6c616e615f73797374656d5f70726f6772616d000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000","r":"0x9cee7b72450461841beb153c65061f02ccfd964e2760913c3f48bd501756d197","s":"0x554779ee71b0f1ec4bb47c5be1afe4ce6cd5907c2b417ff59353094efc448d20","v":1,"creates":null,"chainId":1}, receipt={"to":"0xf92cD566Ea4864356C5491c177A430C222d7e678","from":"0x80253f5106f929d08ab0f69679284ae8A0cCA51c","contractAddress":null,"transactionIndex":299,"gasUsed":{"type":"BigNumber","hex":"0x7d95"},"logsBloom":"0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000","blockHash":"0x5fe004ce79fdda9cde3cdb8adb4f7d7325d7ad01dbe439e8c7a08b93f40d6ebb","transactionHash":"0x9ce3bc6f9a3cff3754254cba691d14728f16bd753d46566dbb96711e5c560f8e","logs":[],"blockNumber":14407004,"confirmations":1,"cumulativeGasUsed":{"type":"BigNumber","hex":"0x01175481"},"effectiveGasPrice":{"type":"BigNumber","hex":"0x0a777c7bf7"},"status":0,"type":2,"byzantium":true}, code=CALL_EXCEPTION, version=providers/5.4.5)
3/17/2022, 7:43:58 PM - Progress exit early with error!

**Reply by Bitclapp** [reply]
@SEJeff Done.. let me know if there's anything else you need.  Thanks.

**Reply by Bitclapp** [reply]
Hi @SEJeff  and @evan-gray  is there any update on this case?  I am a little anxious about how this will turn out.  I would appreciate any additional help or assistance you can provide.  Thank you in advance.

**Reply by Bitclapp** [reply]
Hi @SEJeff  and @evan-gray , Any luck?  Any conclusion?  Any advice on what I can do next?  As usual, your support is highly appreciated. 
Thank you!

**Reply by evan-gray** [reply]
Just tried to get some more details for you. Sounds like some guardians are down on v1 causing the lack of consensus (all the zeros we’re seeing here where signatures would go). Will try to get more info but would be great if you could also reach out to one of the wormhole community managers (admins) on discord so they could track this issue as well and make sure everything is made right 🙂

**Reply by ckeun** [reply]
@Bitclapp can you try redeeming it again via interstellar? I think you should be good to go now

**Reply by Bitclapp** [reply]
Ok thanks a lot to all of you.  I will try it again and let you know how it works.

**Reply by Bitclapp** [reply]
Hello @ckeun @evan-gray and @SEJeff I tried again on interstellar and I have received my wrapped ETH.  Thank you so much!!!!
Can you provide any assistance on unwrapping this ETH?  Thanks again.

**Reply by evan-gray** [reply]
Great to hear! I was actually working on a page for this yesterday 😆  but until it's live, I *think* https://opensea.io/ has an unwrap feature if you click on the wallet on the top right.

**Reply by Bitclapp** [reply]
That page will be very useful but actually I got this token called WWT which does not seem to be the same as WETH...  Not sure what to do with it so I'm stuck.  

Here's the transaction url: https://etherscan.io/token/0x9c30bf2f896dad5f853982f359023a314dbf4c2e?a=0x80253f5106f929d08ab0f69679284ae8a0cca51c

Any guidance on where to convert this to ETH would be nice.

Thanks

**Reply by evan-gray** [reply]
Is it possible that you sent v2 WETH over the v1 bridge or something? 😕

**Comment by ckeun** [comment]
ah shoot. looks like the ETH that you sent from solana --> ethereum was native solana: token address: 7vfCXTUXx5WJV5JADk17DUJ4ksgau7utNKj4b963voxs
@evan-gray is probably right, in that its v2 wrapped WETH.... 
if you could send it back over to solana using the v1 bridge (so sorry) then use https://portalbridge.com/#/transfer ... to send it from solana to ethereum... you should have WETH on ethereum 🥺

**Reply by evan-gray** [reply]
Confirmed this is the case with https://portalbridge.com/#/token-origin-verifier using the address provided above.

**Reply by Bitclapp** [reply]
Thank you so much guys for helping me with this.  So if I understand correctly, I should go back to the v1 bridge (https://v1.wormholebridge.com/#/move?from=ETH&token=SOL) to send those WWT tokens back to Solana.  

When I try to do that, I don't have a WWT option in the list of tokens I can transfer.  I even tried to look up the contract address on v1 but it's not seeing my WWT tokens.  Am I doing something wrong?

Thanks again.

**Reply by ckeun** [reply]
have you tried importing a custom token on your metamask using token address: 0x9C30bF2F896dAD5f853982f359023A314DBF4c2e

**Reply by Bitclapp** [reply]
I've added it to Metamask as per the screenshot below
![image](https://user-images.githubusercontent.com/101778179/160326082-20c49e31-9810-4796-bc11-d84fe86ebc74.png)

When I go to v1 wormhole bridge, I don't see the token in my list of tokens I can transfer.  When I paste the WTT token address nothing happens and there's no search button. 
![image](https://user-images.githubusercontent.com/101778179/160326495-6660ad8b-c5b6-4cf2-8f89-f818c9c2e117.png)

I could be doing something wrong but no idea what it is.  Both my Phantom and Metamask wallets are connected to the v1 wormhole.  

Can you advise on what I can try?  Also, once the WTT arrives on Solana, what kind of token will it be? 

Thanks

**Reply by Bitclapp** [reply]
Hi again @ckeun , I tried passing the parameters via the URL using this: https://v1.wormholebridge.com/#/move?from=ETH&token=WTT 
but v1 Wormhole still does not recognize my WTT token.  Do you have any additional guidance?

**Reply by ckeun** [reply]
yeah i don't know how to troubleshoot this further. i saw your txn in the Proof-of-Assets link and when i click on it, it looks like the token string is set to ETH... 
seems unlikely but maybe try:
https://v1.wormholebridge.com/#/move?from=ETH&token=ETH
@evan-gray, you got any ideas?

**Reply by Bitclapp** [reply]
Well, what you said worked @ckeun .  I used the URL you provided and it detected my WTT.  I was then able to successfully make the transfer to the Solana side.  
![image](https://user-images.githubusercontent.com/101778179/160458923-3edb7e38-783f-41b3-ab39-b7a61481ac6e.png)

Now I have Ether (Wormhole) in my Phantom wallet 
![image](https://user-images.githubusercontent.com/101778179/160459032-da5d1173-aef4-4670-b632-6a8f7e88682b.png)

One question I have now... If I want to go from my ETH (wormhole) to UST on Terra... what are the steps I should take?

**Reply by Bitclapp** [reply]
Oh and I feel so relieved guys, I'm grateful and so happy for all the help and support you provided.  
@evan-gray @SEJeff @ckeun BIG BIG THANK YOU TO YOU!!!!!!

**Reply by ckeun** [reply]
can't believe that worked 😄 
so now you have ETH (wormhole) on solana.... you can either swap it for UST (wormhole) via something like mercurial (https://www.mercurial.finance/swap/ETH-UST) and then use [portalbridge](https://portalbridge.com/#/transfer) to transfer it from solana to terra and you should get native UST.  or you can send the ETH (wormhole) to terra using portalbridge and then go to astroport (https://app.astroport.fi/swap?from=terra14tl83xcwqjy0ken9peu4pjjuu755lrry2uy25r&to=uusd) to get native UST. 

if you need more step by step instructions or anything on using portalbridge, just holler at our support folks on discord or telegram.

**Reply by Bitclapp** [reply]
So happy you went the extra mile and mentioned that as a last resort solution... you could have stopped at "I don't know what to do next".  

In any case, I managed to get UST (wormhole) on Orca.so and moved a small portion to Terra to get native UST.  I ran into issues redeeming but that was because I was using the wrong determination fee.  Once I changed the determination from LUNA to UST everything worked like a charm... and then I went back and transferred a larger amount.  Thanks again!!

**Reply by ckeun** [reply]
glad you got to where you were looking to go! 👏🙌

**Comment by Bitclapp** [comment, marked as answer]
The answer :  https://github.com/certusone/wormhole/discussions/1013#discussioncomment-2454082
